lib/HLSL/DxilDebugInstrumentation.cpp

```diff
@@ -0,0 +1,723 @@
+      uint32_t Opcode : 16;
+    } Details;
+    uint32_t u32Header;
+  } Header;
+  uint32_t UID;
+  uint32_t InstructionOffset;
+};
+
+template< typename ReturnType >
+struct DebugShaderModifierRecordDXILStep : public DebugShaderModifierRecordDXILStepBase {
+  ReturnType ReturnValue;
+};
+
+template< >
+struct DebugShaderModifierRecordDXILStep<void> : public DebugShaderModifierRecordDXILStepBase {
+};
+
+
+uint32_t DebugShaderModifierRecordPayloadSizeDwords(size_t recordTotalSizeBytes) {
+  return ((recordTotalSizeBytes - sizeof(DebugShaderModifierRecordHeader)) / sizeof(uint32_t));
+}
+
+class DxilDebugInstrumentation : public ModulePass {
+
+private:
+  union ParametersAllTogether {
+    unsigned Parameters[3];
+    struct PixelShaderParameters {
+      unsigned X;
+      unsigned Y;
+    } PixelShader;
+    struct VertexShaderParameters {
+      unsigned VertexId;
+      unsigned InstanceId;
+    } VertexShader;
+    struct ComputeShaderParameters {
+      unsigned ThreadIdX;
+      unsigned ThreadIdY;
+      unsigned ThreadIdZ;
+    } ComputeShader;
+  } m_Parameters = { 0,0,0 };
+
+  union SystemValueIndices {
+    struct PixelShaderParameters {
+      unsigned Position;
+    } PixelShader;
+    struct VertexShaderParameters {
+      unsigned VertexId;
+      unsigned InstanceId;
+    } VertexShader;
+  };
+
+  uint64_t m_UAVSize = 1024*1024;
+  Value * m_SelectionCriterion = nullptr;
+  CallInst * m_HandleForUAV = nullptr;
+  Value * m_InvocationId = nullptr;
+
+  // Together these two values allow branchless writing to the UAV. An invocation of the shader
+  // is either of interest or not (e.g. it writes to the pixel the user selected for debugging
+  // or it doesn't). If not of interest, debugging output will still occur, but it will be
+  // relegated to the very top few bytes of the UAV. Invocations of interest, by contrast, will
+  // be written to the UAV at sequentially increasing offsets.
+
+  // This value will either be one or zero (one if the invocation is of interest, zero otherwise)
+  Value * m_OffsetMultiplicand = nullptr;
+  // This will either be zero (if the invocation is of interest) or (UAVSize)-(SmallValue) if not.
+  Value * m_OffsetAddend = nullptr;
+
+  Constant * m_OffsetMask = nullptr;
+
+  std::map<uint32_t, Value *> m_IncrementInstructionBySize;
+
+  unsigned int m_InstructionIndex = 0;
+
+  struct BuilderContext {
+    Module &M;
+    DxilModule &DM;
+    LLVMContext & Ctx;
+    OP * HlslOP;
+    IRBuilder<> & Builder;
+  };
+
+  uint32_t m_RemainingReservedSpaceInBytes = 0;
+  Value * m_CurrentIndex = nullptr;
+
+public:
+  static char ID; // Pass identification, replacement for typeid
+  explicit DxilDebugInstrumentation() : ModulePass(ID) {}
+  const char *getPassName() const override { return "DXIL Force Early Z"; }
+  void applyOptions(PassOptions O) override;
+  bool runOnModule(Module &M) override;
+
+private:
+  SystemValueIndices addRequiredSystemValues(BuilderContext & BC);
```

Comment:
If you need to touch this up anyway - code convention in this part of the codebase is "BuilderContext &BC", with no space between ampersand and 'BC'

Reply:
fixed... (and a couple of other things like it)